libraries/AP_Mission/AP_Mission.h

```diff
@@ -555,6 +560,7 @@ class AP_Mission {
     AP_Int16                _cmd_total;  // total number of commands in the mission
     AP_Int8                 _restart;   // controls mission starting point when entering Auto mode (either restart from beginning of mission or resume from last command run)
     AP_Int16                _options;    // bitmask options for missions, currently for mission clearing on reboot but can be expanded as required
+    AP_Int32                _max_nav_dist; //maximum allowed distance to first waypoint
```

Comment:
Probably worth considering the type here.  int32 gives a rather too-large range.  Is 65km too little?